lib/jekyll/static_file.rb

```diff
@@ -144,7 +144,11 @@ def copy_file(dest_path)
       if @site.safe || Jekyll.env == "production"
         FileUtils.cp(path, dest_path)
       else
-        FileUtils.copy_entry(path, dest_path)
+        begin
+          FileUtils.ln(path, dest_path)
+        rescue
```

Comment:
HAVE_LINK only decides whether they throw an unavailable error.  It doesn't make it easier to test if symlinks are explicitly supported (that is unless you create a helper that does that out side of this context and tests for that failure directly.)

If your worry is x-links then that's a problem that needs to be solved with a configuration option, not an implicit option that drags down performance without public cause.  I still consider that a user problem but you are right that is a case that need be addressed but to me, the point of least resistance in acceptance is configuring hard links in explicitly and old behavior implicitly.  This way nobody is caught off guard, and we aren't to explain why it's not working when we claim it will.